src/editor/window.rs

```diff
@@ -255,16 +255,17 @@ impl Window {
         rows: i64,
         cols: i64,
     ) {
-        let mut top_to_bottom;
-        let mut bottom_to_top;
-        let y_iter: &mut dyn Iterator<Item = i64> = if rows > 0 {
-            top_to_bottom = (top as i64 + rows)..bottom as i64;
-            &mut top_to_bottom
-        } else {
-            bottom_to_top = (top as i64..(bottom as i64 + rows)).rev();
-            &mut bottom_to_top
-        };
+        self.grid.scroll_region(
+            top as usize,
+            bottom as usize,
+            left as usize,
+            right as usize,
+            rows as isize,
+            cols as isize,
+        );
 
+        // Scrolls must not only translate the rendered texture, but also must move the grid data
```

Comment:
This comment is confusing now because it is placed after the actual data move. Maybe move it to the top of the section and reword it to say:

"Scrolls must move the data and send a WindowDrawCommand to move the rendered texture so that future renders draw correctly"